### DISCUSSION TITLE
Is it possible to have ADetailer run before upscaling?

### DISCUSSION THREAD

**Discussion by Jonathanese** [Bing-su/adetailer#384] [General]
I generally create images at 512x768 and then do a 4x upscale with 4xUltrasharpV10.

I notice that once an image is done generating, the compositional quality is somewhat improved, but every place that ADetailer touches seems to be based on the initial 512x768 resolution, assuming it's pulling these values directly from the UI.

Rather than trying to run ADetailer at 2048x3072, I'd like to be able to run ADetailer on the 512x768 image, and then upscale the resulting image. This works relatively well doing so manually. And the nice thing is that it would produce a better-blended composition as a result.

**Comment by Bing-su** [comment]
It's not possible with the current structure of adetailer.

**Reply by Jonathanese** [reply]
Thanks!

I basically stopped using the integrated upscaler and switched to the ComfyAI addon. So now I do everything at low res, and upscale as a post-process using the upscaling loader and node. Works like a charm!

**Reply by Woisek** [reply]
Thanks for mentioning this. Could you maybe share this workflow, as I'm also interested in this, and want to learn the functionality of it? 🙂👍
